packages/glimmer-runtime/tests/initial-render-test.ts

```diff
@@ -349,6 +349,46 @@ test("The compiler can handle simple helpers", function() {
   compilesTo('<div>{{testing title}}</div>', '<div>hello</div>', { title: 'hello' });
 });
 
+test("GH#13999 The compiler can handle simple helpers with inline null parameter", function() {
+  env.registerHelper('say-hello', function() {
+    return 'hello';
+  });
+
+  compilesTo('<div>{{say-hello null}}</div>', '<div>hello</div>');
+});
+
+test("GH#13999 The compiler can handle simple helpers with inline undefined parameter", function() {
+  env.registerHelper('say-hello', function() {
+    return 'hello';
+  });
+
+  compilesTo('<div>{{say-hello undefined}}</div>', '<div>hello</div>');
+});
+
+test("GH#13999 The compiler can handle simple helpers with inline undefined parameter", function() {
+  env.registerHelper('say-hello', function() {
+    return 'hello';
+  });
+
+  compilesTo('<div>{{say-hello undefined}}</div>', '<div>hello</div>');
+});
+
+test("GH#13999 The compiler can handle components with null named arguments", function() {
+  env.registerHelper('say-hello', function() {
+    return 'hello';
+  });
+
+  compilesTo('<div>{{say-hello foo=null}}</div>', '<div>hello</div>');
+});
+
+test("GH#13999 The compiler can handle components with undefined named arguments", function() {
+  env.registerHelper('say-hello', function() {
+    return 'hello';
+  });
+
+  compilesTo('<div>{{say-hello foo=undefined}}</div>', '<div>hello</div>');
+});
+
```

Comment:
I can think of at least a few more cases to test – `{{null}}` ("append"/content position), `<div class="foo {{null}} bar" />` (concat) and `{{#if null}}` ("block" arguments).